clang/include/clang/Basic/Attr.td

```diff
@@ -1228,7 +1228,7 @@ def SYCLGlobalVar : InheritableAttr {
 def SYCLKernel : InheritableAttr {
   let Spellings = [Clang<"sycl_kernel">];
   let Subjects = SubjectList<[FunctionTmpl]>;
-  let LangOpts = [SYCLIsHost, SYCLIsDevice];
+  let LangOpts = [SYCLIsDevice];
```

Comment:
Should this be using `SilentlyIgnoreSYCLIsHost` instead?

Reply:
I wasn't sure if going from a defined behavior on the host to silently being ignored gave proper notice.  May be do this some time in the future?

Reply:
I believe we use `SilentlyIgnoreSYCLIsHost` to avoid confusion that arises with diagnostics because compilation model has multiple phases. When user compiles the code using driver (i.e. device compilation + host compilation + other phases we have now), this warning will be confusing I think. From a user perspective here, the attribute was not ignored. It was used in device compilation

Reply:
I didn't think typical users would see this warning since they don't directly use this attribute, but through CL/sycl.hpp interfaces, and we suppress diagnostics coming from system header files.

Reply:
@turinevgeny, can you share how users might have used this attribute on host?   Would that also have been through supplied header file interfaces?  In which case, I can change the LangOpts to SilentlyIgnoreSYCLIsHost, since it wouldn't matter there either.

Reply:
@premanandrao There was a plan to use it on host through some header interfaces, but now there are no such plans.

Reply:
Thanks @turinevgeny.   I have made this attribute SilentlyIgnoreSYCLIsHost.